mitodl/open-discussions
# avoid parsing all documents at once

#### Pre-Flight checklist

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3436
closes https://github.com/mitodl/open-discussions/issues/3439

#### What's this PR do?
Some of our courses have thousands of course documents, for example PROD/8/8.01/Fall_2016/8-01sc-classical-mechanics-fall-2016/

This PR updates the ocw update and reindex code to avoid loading all the course document data into memory at the same time.

#### How should this be manually tested?
Verify that the backpopulate_ocw_data, backpopulate_ocw_files, backpopulate_xpro_data and recreate_index --course commands run without errors
